### PR TITLE
[ch9800] provide 400 error when Add New Audit Staff Team Member

### PR DESCRIPTION
**Pull request by domdinicola**
[ch9800]

### CODE REVIEW

File: src/etools/applications/audit/serializers/auditor.py [hunk 1]
```diff
@@ -40,10 +40,15 @@ def validate(self, attrs):
         validated_data = super().validate(attrs)
         user_pk = validated_data.pop('user_pk', None)
 
-        if not self.instance:
+        if self.instance and user_pk:
```
Comment (reinbach) [inline src/etools/applications/audit/serializers/auditor.py]: I think we can clean up these `if/else` statements a bit more. The `user_pk` is being used all over.

Maybe the following may be clearer;
```
if user_pk:
  if self.instance:
    ...
  else:
   ...
else:
  if "user" not in validated_data:
   ...
```

Comment (domdinicola) [reply src/etools/applications/audit/serializers/auditor.py]: sure

Comment (domdinicola) [reply src/etools/applications/audit/serializers/auditor.py]: @reinbach you comment hsa been addressed

File: src/etools/applications/audit/serializers/auditor.py [hunk 2]
```diff
@@ -40,15 +40,14 @@ def validate(self, attrs):
...
+        if user_pk:
+            if hasattr(user_pk, 'purchase_order_auditorstaffmember'):
+                firm = user_pk.purchase_order_auditorstaffmember.auditor_firm
+                raise serializers.ValidationError({'user': _('User is already assigned to ') + str(firm)})
+            if not self.instance:
                 validated_data['user'] = user_pk
-
-            if 'user' not in validated_data and not user_pk:
-                raise serializers.ValidationError({'user': _('This field is required.')})
+        elif 'user' not in validated_data and not user_pk:
```
Comment (reinbach) [inline src/etools/applications/audit/serializers/auditor.py]: You probably don't need the `and not user_pk` part as you should not get here based on the initial `if user_pk`

**Pull request merged**